### PR TITLE
Dashboard: Only show cache timeout option in panel edit if enabled in data source plugin json

### LINKED ISSUE #24010: QueryOptions: Hide Cache timeout option
Cache timeout option should only be shown in the query options if data source plugin meta.queryOptions.cacheTimeout is true

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #24010

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/panel_editor/QueryOptions.tsx [hunk 1]
```diff
@@ -300,7 +300,7 @@ export class QueryOptions extends PureComponent<Props, State> {
       >
         {this.renderMaxDataPointsOption()}
         {this.renderIntervalOption()}
-        {this.renderCacheTimeoutOption()}
+        {this.props.datasource.meta.queryOptions.cacheTimeout && this.renderCacheTimeoutOption()}
```
Comment (torkelo) [inline public/app/features/dashboard/panel_editor/QueryOptions.tsx:303]: think queryOptions is optional, this should cause increase in strictNull errors,  is there we can move this condition into renderCacheTimeoutOption ?

Comment (peterholmberg) [reply public/app/features/dashboard/panel_editor/QueryOptions.tsx:303]: Oh, yes can have a look

File: public/app/features/dashboard/panel_editor/QueryOptions.tsx [hunk 2]
```diff
@@ -130,7 +130,7 @@ export class QueryOptions extends PureComponent<Props, State> {
     const tooltip = `If your time series store has a query cache this option can override the default cache timeout. Specify a
     numeric value in seconds.`;
 
-    if (!datasource.meta.queryOptions?.maxDataPoints) {
+    if (!datasource.meta.queryOptions?.maxDataPoints || !datasource.meta.queryOptions?.cacheTimeout) {
```
Comment (torkelo) [inline public/app/features/dashboard/panel_editor/QueryOptions.tsx:133]: this is wrong (my mistake), remove the datasource.meta.queryOptions?.maxDataPoints

Comment (torkelo) [inline public/app/features/dashboard/panel_editor/QueryOptions.tsx:133]:
```suggestion
    if (!datasource.meta.queryOptions?.cacheTimeout) {
```

**Pull request merged**